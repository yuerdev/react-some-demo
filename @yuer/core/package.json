{
  "name": "@yuer/core",
  "version": "0.1.0",
  "dependencies": {
    "rxjs": "^7.5.6"
  },
  "files": [
    "dist/"
  ],
  "exports": {
    ".": {
      "require": "./dist/index.cjs",
      "import": "./dist/index.mjs"
    },
    "./index.ts": "./index.ts"
  },
  "type": "module",
  "types": "./dist/index.d.ts",
  "main": "./dist/index.cjs",
  "module": "./dist/index.mjs"
}