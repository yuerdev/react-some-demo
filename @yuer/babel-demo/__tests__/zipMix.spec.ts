import { transformSync } from "@babel/core";
import { dirname } from "path";
import { fileURLToPath } from "url";
import Mix from "../mix";
import Zip from "../zip";

const __dirname = dirname(fileURLToPath(import.meta.url));

const compileToSnapshot = (code: string) => {
    const t = transformSync(code, {
        root: __dirname,
        parserOpts: {
            plugins: ["jsx"],
        },
        plugins: [[Zip], [Mix]],
        generatorOpts: {
            comments: false,
            compact: true
        }
    });

    return `
${code}
    
↓ ↓ ↓ ↓ ↓ ↓

${t?.code}  
`;
};

describe("zip mix test", () => {
    it("zip mix", () => {
        const result = compileToSnapshot(`
        function func() {
            const num1 = 1;
            const num2 = 2;
            const num3 = /*@__PURE__*/add(1, 2);
            const num4 = add(3, 4);
            console.log(num2);
            return num2;
            console.log(num1);
            function add (aaa, bbb) {
                return aaa + bbb;
            }
        }
        func();
    `);

        expect(result).toMatchSnapshot();
    });

});