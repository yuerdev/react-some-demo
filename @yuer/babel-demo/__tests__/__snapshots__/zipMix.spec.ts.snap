// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`zip mix test zip mix 1`] = `
"

        function func() {
            const num1 = 1;
            const num2 = 2;
            const num3 = /*@__PURE__*/add(1, 2);
            const num4 = add(3, 4);
            console.log(num2);
            return num2;
            console.log(num1);
            function add (aaa, bbb) {
                return aaa + bbb;
            }
        }
        func();
    
    
↓ ↓ ↓ ↓ ↓ ↓

function _g(){const _c=1;const _d=2;_f(3,4);console.log(_d);return _d;function _f(_a,_b){return _a+_b;}}_g();  
"
`;
